src/classes/HouseholdAccounts.cls

```diff
@@ -98,4 +157,362 @@ public inherited sharing class HouseholdAccounts {
+    }
+
+    /*******************************************************************************************************
+    * @description Updates an HH Account's naming, primary contact, and rollups
+    * @param accountIds the list of Account (Id's) that need updating.
+    */
+    public static void updateHouseholds(List<Id> accountIds, TDTM_Runnable.DmlWrapper dmlWrapper) {
+        if (accountIds.isEmpty()) {
+            return;
+        }
+
+        resetPrimaryContactForAccount(accountIds, dmlWrapper);
+        renameHHAccounts(accountIds);
+        rollupAccounts(accountIds);
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of Household or Account Ids, update their Household Names
+    * @param accountIds the list of Household or Account Ids that need updating.
+    * @return void
+    */
+    public static void renameHHAccounts(List<Id> accountIds) {
+        if (accountIds.isEmpty()) {
+            return;
+        }
+
+        if (shouldProcessSynchronously(accountIds)) {
+            HouseholdNamingService householdNamingService = new HouseholdNamingService();
+            householdNamingService.updateHouseholdNameAndMemberCount(accountIds);
+        } else {
+            householdNamingService.updateHouseholdNameAndMemberCountAsync(accountIds);
+        }
+    }
+
+    private static Boolean shouldProcessSynchronously(List<Id> accountIds) {
+        return accountIds.size() == 1 || System.isFuture() || System.isBatch() || System.isQueueable();
+    }
+
+    /*******************************************************************************************************
+    * @description sets the Primary Account for each Account in the list of Accounts
+    * @param accountIds the list of Account (Id's) that need updating.
+    */
+    private static void resetPrimaryContactForAccount(List<Id>
+            accountIds, TDTM_Runnable.DmlWrapper dmlWrapper) {
+
+        // for each account, we need a set of their contacts.
+        Map<Id, Set<Id>> contactIdsByAccountId = new Map<Id, Set<Id>>();
+
+        for (Contact contactRecord : [
+                SELECT Id, AccountId, Account.npe01__One2OneContact__c
+                FROM Contact
+                WHERE AccountId IN :accountIds
+        ]) {
+            Set<Id> contactIds = contactIdsByAccountId.get(contactRecord.AccountId);
+            if (contactIds == null) {
+                contactIds = new Set<Id>();
+                contactIdsByAccountId.put(contactRecord.AccountId, contactIds);
+            }
+            contactIds.add(contactRecord.Id);
+        }
+
+        // now go thru each account and update their primary contact if the current one isn't valid
+        List<Account> accounts = new List<Account>();
+        for (Account accountRecord : [SELECT Id, npe01__One2OneContact__c FROM Account WHERE Id IN :accountIds]) {
```

Comment:
I feel like these records should already be available?

Reply:
I think we can combine these queries, my only reservation was "What happens when one of the Accounts has no Contacts?". When looping through the second query, contactIds would be null for any Account that had zero contacts, and that looks to be OK. 

We'll still need a second loop here, but we wouldn't need to query Accounts. We can gather all of the AccountIds in the first loop w/ their `npe01__One2OneContact__c`, and add those to the Accounts to update if the Contact Id in `npe01__One2OneContact__c` does not match with any of the Contacts that are currently part of the given Household.

Reply:
I like the plan, Ben, adding it to the list of items to follow up on on this class: https://salesforce.quip.com/n2vzAQOiRbT7#UMTACAd3ibZ